### PR TITLE
feat(suite): create View-only feature flag in settings

### PR DESCRIPTION
**Pull request by jvaclavik**
## Related Issue

Resolve 

## Screenshots:
image

### CODE REVIEW

File: packages/suite/src/views/settings/SettingsDebug/ViewOnly.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import { Checkbox } from '@trezor/components';
+import { isDesktop } from '@trezor/env-utils';
+import { useDispatch, useSelector } from 'src/hooks/suite';
+import { ArrayElement } from '@trezor/type-utils';
+
+import { setDebugMode } from 'src/actions/suite/suiteActions';
+import { DebugModeOptions } from 'src/reducers/suite/suiteReducer';
+import { ActionColumn, SectionItem, TextColumn } from 'src/components/suite';
+
+type TransportMenuItem = {
```
Comment (peter-sanderson) [inline packages/suite/src/views/settings/SettingsDebug/ViewOnly.tsx]: This seems to be unused

Comment (peter-sanderson) [reply packages/suite/src/views/settings/SettingsDebug/ViewOnly.tsx]: Probably forgotten copy-paste

Comment (jvaclavik) [reply packages/suite/src/views/settings/SettingsDebug/ViewOnly.tsx]: Good point, fixed

File: packages/suite/src/views/settings/SettingsDebug/ViewOnly.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    );
+    const dispatch = useDispatch();
+
+    return (
+        <SectionItem>
+            <TextColumn title={`Show redesigned view-only`} />
+            <ActionColumn>
+                <Checkbox
+                    isChecked={isViewOnlyModeVisible}
+                    onClick={() => {
```
Comment (peter-sanderson) [inline packages/suite/src/views/settings/SettingsDebug/ViewOnly.tsx:19]: optional/nit: extract to named function for better readability

**Review by peter-sanderson — APPROVED** [review]

**Pull request merged**